pkg/storage/stores/shipper/util/util.go

```diff
@@ -252,3 +255,21 @@ func ListDirectory(ctx context.Context, dirName string, objectClient chunk.Objec
 	}
 	return objects, nil
 }
+
+func QueryKey(q chunk.IndexQuery) string {
+	ret := q.TableName + sep + q.HashValue
+
+	if len(q.RangeValuePrefix) != 0 {
+		ret += sep + yoloString(q.RangeValuePrefix)
```

Comment:
Does yolostring gain us much here? We are still doing allocations to do the string concatenation and I'm worried this leaves us vulnerable to mistakes in the future. 

Same concern with the use below.

Reply:
Yeah, you are right; we won't gain much here. I will remove it.
Thanks!